src/main/java/com/gtnewhorizons/angelica/glsm/Feature.java

```diff
@@ -0,0 +1,245 @@
+            // GL_MAP_COLOR and GL_MAP_STENCIL flags
+            // GL_ZOOM_X and GL_ZOOM_Y factors
+            // GL_READ_BUFFER setting
+        );
+        attribToFeatures.put(GL11.GL_POINT_BIT, FeatureSet.newMap()
+            // GL_POINT_SMOOTH flag
+            // Point size
+        );
+        attribToFeatures.put(GL11.GL_POLYGON_BIT, FeatureSet.newMap()
+            .put(GLStateManager.cullState) // GL_CULL_FACE enable bit
+            .put(GLStateManager.cullState)// GL_CULL_FACE_MODE value
+            // GL_FRONT_FACE indicator
+            // GL_POLYGON_MODE setting
+            // GL_POLYGON_SMOOTH flag
+            // GL_POLYGON_STIPPLE enable bit
+            // GL_POLYGON_OFFSET_FILL flag
+            // GL_POLYGON_OFFSET_LINE flag
+            // GL_POLYGON_OFFSET_POINT flag
+            // GL_POLYGON_OFFSET_FACTOR
+            // GL_POLYGON_OFFSET_UNITS
+        );
+        attribToFeatures.put(GL11.GL_POLYGON_STIPPLE_BIT, FeatureSet.newMap()
+            // Polygon stipple pattern
+        );
+        attribToFeatures.put(GL11.GL_SCISSOR_BIT, FeatureSet.newMap()
+            // GL_SCISSOR_TEST enable bit
+            // Scissor box
+        );
+        attribToFeatures.put(GL11.GL_STENCIL_BUFFER_BIT, FeatureSet.newMap()
+            // GL_STENCIL_TEST enable bit
+            // Stencil function and reference value
+            // Stencil value mask
+            // Stencil fail, pass, and depth buffer pass actions
+            // Stencil buffer clear value
+            // Stencil buffer writemask
+        );
+        attribToFeatures.put(GL11.GL_TEXTURE_BIT, FeatureSet.newMap()
+            // Enable bits for the four texture coordinates
+            // Border color for each texture image
+            // Minification function for each texture image
+            // Magnification function for each texture image
+            // Texture coordinates and wrap mode for each texture image
+            // Color and mode for each texture environment
+            // Enable bits GL_TEXTURE_GEN_x, x is S, T, R, and Q
+            // GL_TEXTURE_GEN_MODE setting for S, T, R, and Q
+            // glTexGen plane equations for S, T, R, and Q
+        );
+        attribToFeatures.put(GL11.GL_TRANSFORM_BIT, FeatureSet.newMap()
+            // Coefficients of the six clipping planes
+            // Enable bits for the user-definable clipping planes
+            .put(GLStateManager.matrixMode)
+            // GL_NORMALIZE flag
+            // GL_RESCALE_NORMAL flag
+        );
+        attribToFeatures.put(GL11.GL_VIEWPORT_BIT, FeatureSet.newMap()
+            // Depth range (near and far)
+            .put(GLStateManager.viewportState)
+        );
+    }
+
+    public static Set<IStateStack<?>> getFeatures(int attrib) {
+        return attribToFeatures.getOrDefault(attrib, Collections.emptySet());
+    }
+
+    private static class FeatureSet<K> extends HashSet<K> {
+
+        public static FeatureSet<IStateStack<?>> newMap() {
```

Comment:
Could probably just use `ImmutableSet.of(...)` for this stuff instead of a HashSet subclass.

Reply:
Perhaps. Turned out smaller than anticipated